includes/Grrr/SimplyStaticDeploy/Dependencies/ConfigStructureDependency.php

```diff
@@ -0,0 +1,30 @@
+<?php namespace Grrr\SimplyStaticDeploy\Dependencies;
+
+use Grrr\SimplyStaticDeploy\Config\RequiredFields;
+
+class ConfigStructureDependency implements DependencyInterface {
+
+    private $constName;
+
+    public function __construct(string $constName) {
+        $this->constName = $constName;
+    }
+
+    public function is_met(): bool {
+        return empty(RequiredFields::getMissingOptions(constant($this->constName)));
```

Comment:
This will throw when `SIMPLY_STATIC_DEPLOY_CONFIG` is not defined.

Reply:
True. Fixed it with a `defined` check.

Reply:
Quickly checked this locally to test the whole thing all out, but the message below also relies on this constant, sorry 😇

Reply:
Curses. 

Yeah, I sort of assumed that this would only ever be used _after_ that other dependency check. But assumptions are the mother of all fuckups, so there you go.

Reply:
This got me thinking: _why_ is the ConfigStructure thing blowing up for you?
But the `reduce` call in `are_met()` is not short-circuited, meaning all dependencies are checked, even when a previous one already failed.

I changed this in 5d960cb. @HammenWS you should also take a look, this is a handy trick for cases where you don't want to finish all iterations when you know you failed already.
So every code that's used to check whether _all_ things are in a certain state, or for instance when you need to find a single item in a collection, could all benefit from `f\reduced`.